gdrive_sync/api.py

```diff
@@ -339,17 +340,27 @@ def walk_gdrive_folder(folder_id: str, fields: str) -> Iterable[Dict]:
                 yield sub_result
 
 
+def get_pdf_title(drive_file: DriveFile) -> str:
+    """Get the title of a PDF from its metadata, if available"""
+    pdf_file = io.BytesIO(GDriveStreamReader(drive_file).read())
```

Comment:
I used `docker stats` to monitor memory usage on celery, and synced 40 pdfs (22 MB each), results are:

with current code: memory usage increased from ~500 MB to 2 GB
with no pdf reading: from ~500 MB to 850 MB
with the code change below: from ~500 MB to 1.4 GB

```python
    with io.BytesIO(GDriveStreamReader(drive_file).read()) as pdf_file:
        pdf_reader = PyPDF2.PdfReader(pdf_file)
        pdf_metadata = pdf_reader.metadata
        pdf_file.close()
        if "/Title" in pdf_metadata and pdf_metadata["/Title"] != "":
            return pdf_metadata["/Title"]
        return drive_file.name
```

1.4 GB. is still high, but better than 2 GB, so I think it might be worth making the change above.

Reply:
just tweaked the code snippet above a bit